pkg/xray/resource_xray_repository_config.go

```diff
@@ -0,0 +1,349 @@
+		}
+
+		m := map[string]interface{}{}
+		m["vuln_contextual_analysis"] = repoConfig.VulnContextualAnalysis
+		m["retention_in_days"] = repoConfig.RetentionInDays
+
+		return []interface{}{m}
+	}
+
+	var packAllOtherArtifacts = func(otherArtifacts AllOtherArtifacts) []interface{} {
+		m := map[string]interface{}{
+			"index_new_artifacts": otherArtifacts.IndexNewArtifacts,
+			"retention_in_days":   otherArtifacts.RetentionInDays,
+		}
+
+		return []interface{}{m}
+	}
+
+	var packPatterns = func(patterns []Pattern) []interface{} {
+		var ps []interface{}
+
+		for _, pattern := range patterns {
+			p := map[string]interface{}{
+				"include":             pattern.Include,
+				"exclude":             pattern.Exclude,
+				"index_new_artifacts": pattern.IndexNewArtifacts,
+				"retention_in_days":   pattern.RetentionInDays,
+			}
+
+			ps = append(ps, p)
+		}
+
+		return ps
+	}
+
+	var packRepoPathsConfigList = func(repoPathsConfig *RepoPathsConfiguration) []interface{} {
+		if repoPathsConfig == nil {
+			return []interface{}{}
+		}
+
+		m := map[string]interface{}{}
+		m["pattern"] = packPatterns(repoPathsConfig.PatternList)
+		m["all_other_artifacts"] = packAllOtherArtifacts(repoPathsConfig.OtherArtifacts)
+
+		return []interface{}{m}
+	}
+
+	var packRepositoryConfig = func(ctx context.Context, repositoryConfig RepositoryConfiguration, d *schema.ResourceData) diag.Diagnostics {
+		if err := d.Set("repo_name", repositoryConfig.RepoName); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("repo_config", packGeneralRepoConfig(repositoryConfig.RepoConfig)); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("repo_paths_config", packRepoPathsConfigList(repositoryConfig.RepoPathsConfig)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		return nil
+	}
+
+	var resourceXrayRepositoryConfigRead = func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		repositoryConfig := RepositoryConfiguration{}
+
+		resp, err := m.(*resty.Client).R().SetResult(&repositoryConfig).
+			SetPathParams(map[string]string{
+				"repo_name": d.Id(),
+			}).
+			Get("xray/api/v1/repos_config/{repo_name}")
+
+		if err != nil {
+			if resp != nil && resp.StatusCode() != http.StatusOK {
+				tflog.Error(ctx, fmt.Sprintf("Request payload is invalid as repo (%s) is either not indexed or does not exist", d.Id()))
+				d.SetId("")
+			}
+			return diag.FromErr(err)
+		}
+
+		return packRepositoryConfig(ctx, repositoryConfig, d)
+	}
+
+	var resourceXrayRepositoryConfigCreate = func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		repositoryConfig := unpackRepositoryConfig(d)
+
+		resp, err := m.(*resty.Client).R().SetBody(&repositoryConfig).Put("xray/api/v1/repos_config")
+		if err != nil {
+			if resp != nil && resp.StatusCode() != http.StatusOK {
+				tflog.Error(ctx, fmt.Sprintf("Request payload is invalid as repo (%s) is either not indexed or does not exist", d.Id()))
```

Comment:
Why is this error log special?

Reply:
Why do you think it's special? :) Just a regular non-special log.

Reply:
We only log if there's error and the response is not 200. What about there is error (e.g. networking error before HTTP request is made) but no response? And logging here is really for us, not for the users.

Reply:
I've removed the log and just return `diag.FromErr(err)`, should be enough.